aws/cloudformation/components/database.yml.erb

```diff
@@ -53,7 +53,7 @@
       Description: !Sub "Database credentials for ${AWS::StackName} writer user."
 <% if database -%>
       GenerateSecretString:
-        SecretStringTemplate: !Sub '{"username": "application-writer"}'
+        SecretStringTemplate: '{"username": "writer"}'
```

Comment:
nothing is using these old usernames?

Reply:
The only components that utilize the SQL login credentials provisioned in this template are RDS Proxy and the `bin/mysql-client-` scripts. The `mysql-client-` reader & writer scripts currently do not work because the passwords generated by CloudFormation when it provisioned the AWS Secrets Manager Secret Resources in this template are different from the actual passwords of the `readonly` and `application-writer` SQL users.

Deploying this change will NOT change the `readonly` and `application-writer` SQL users that our web application servers and cron jobs utilize, nor does it change the Chef Attributes and the `CDO.` configuration settings where our Ruby code retrieves those credentials.

It will change:

- Anyone connecting to the Reporting RDS Proxy endpoint as the user `readonly` will be rejected. They'll either need the new `reader` credentials, or to use the `mysql-client-dashboard-reporting` script
- All of the `bin/mysql-client-*` scripts will now work. This the only Ruby code that utilizes Stack-specific Secrets to retrieve database connection credentials.